snuba/query/timeseries_extension.py

```diff
@@ -122,3 +122,48 @@ def __init__(
             ),
             processor=TimeSeriesExtensionProcessor(timestamp_column),
         )
+
+
+class SessionsTimeSeriesExtensionProcessor(TimeSeriesExtensionProcessor):
+    def __init__(
+        self,
+        timstamp_column: str,
+        granularity_check_func: Optional[Callable[[Query], Any]],
+    ) -> None:
+        super(SessionsTimeSeriesExtensionProcessor, self).__init__(timstamp_column)
+        if granularity_check_func is not None:
```

Comment:
Why is this optional? The process_query function will fail if this is None.

Reply:
I don't understand what you mean by `granularity` field is not a source of truth

I need the granularity field because, for `time_window` <= 1h, I want to use the raw sessions table and for >1h I want to use the materialized.  I extract it in the Processor because I need the time window to decide on that and I don't have the time_window in the `select_storage` function

Reply:
>Why is this optional? The process_query function will fail if this is None.

Sure I will add a noop